// js/resultats.js
var fs=require('fs');
var points={D:3,P:3,E:5,T:2,N:0}
var liste=[]
var dico={}


var init = (function() {

})();

exports.calcul=function(day,jours,scores) {
	liste=[]
	var feuille1={},
		feuille2={}
	for(var i=0;i<jours.length;i++) {
		match=jours[i]
		feuille1=initcalc(match[0],match[1])
		feuille1.jour=day
		feuille1=calcPhase1(feuille1, scores.getScores(day.toString()+','+match[0].toString()))
		//console.log(feuille)
		feuille2=initcalc(match[1],match[0])
		feuille2.jour=day
		feuille2=calcPhase1(feuille2, scores.getScores(day.toString()+','+match[1].toString()))
		feuille1.contre=feuille2.pour
		feuille2.contre=feuille1.pour
		feuille1.essctr=feuille2.nbEssai
		feuille2.essctr=feuille1.nbEssai
		feuille1=calcPhase2(feuille1)
		feuille2=calcPhase2(feuille2)
		liste.push(feuille1);
		liste.push(feuille2);
	}
	//console.log(liste)
	fs.writeFileSync('./data/RESJR/'+day.toString()+'.json',JSON.stringify(liste,null,4));
	makeDico()
}

function initcalc(eq,adv) {
	var feuille={jour:"",team:eq,adv:adv,pts:0,nbMatch:0,bo:0,
				bd:0,pour:0,contre:0,gagne:0,neutre:0,perd:0,
				nbEssai:0, nbDrop:0, nbTransf:0, nbPenalite:0, dif:0, essctr:0}
	return feuille
}

function calcPhase1(feuille, scores) {
	//console.log(scores)
	for (var i=0;i<scores.length;i++) {
		sc = scores[i]
		feuille.pour+=points[sc.pts]
		feuille.nbMatch=1
		if (sc.pts=="D")
			feuille.nbDrop+=1;
		if (sc.pts=="E")
			feuille.nbEssai+=1;
		if (sc.pts=="T")
			feuille.nbTransf+=1;
		if (sc.pts=="P")
			feuille.nbPenalite+=1;
	}
	return feuille
}

function calcPhase2(feuille) {
	if (feuille.nbMatch!=0) {
		feuille.dif=feuille.pour-feuille.contre
		if (feuille.pour == feuille.contre) {
			feuille.pts=2
			feuille.neutre=1
		}
		if (feuille.pour > feuille.contre) {
			feuille.pts=4
			feuille.gagne=1
		}
		if (feuille.pour < feuille.contre) {
			feuille.perd=1
			if (feuille.dif >= -7 && feuille.dif <0) {
				feuille.bd=1
				feuille.pts+=1
			}
		}
		if (feuille.nbEssai - feuille.essctr >=3 ) {
			feuille.bo=1
			feuille.pts+=1
		}
	}
	return feuille
}

exports.listerResultats=function() {
	return liste
}

function makeDico() {
	for (var i=0; i<liste.length; i++) {
		item = liste[i]
		dico[item.team]=item.pour
	}
	//console.log(dico)

}
exports.getPts=function(equipe) {
	return dico[equipe]
}
