// js/teams.js
var fs=require('fs');
var liste=""
var teams=[]
var file='./data/teams.json'

var init = (function() {
	var content = fs.readFileSync(file, 'utf8');
		content = JSON.parse(content);
		teams = content
		initialize();
		/*
		//console.log(content);
		var buf = [];
		buf.push("<html><head></head><body><h1>liste des teams</h1>");
		buf.push("<table border='1'>");
		buf.push("<tr><th>Action</th><th>Code</th><th>Nom</th><th>Lieu</th></tr>");
		//buf.push(content);
		var t;
		for (i=0; i < content.length; i++) {
			t = content[i];
			buf.push("<tr><td><a href='/team/sup/"+i+"'>sup</a> - <a href='/team/mod/"+i+"'>mod</a></td><td>"
				+t.code+"</td><td>"+t.nom+"</td><td>"+t.lieu+"</td></tr>");
			//buf.push(t);
		}
		buf.push("</table>");
		buf.push("<hr><a href='/'>Retour</a>");
		buf.push("<a href='/newteam'>Nouveau Team</a>");
		buf.push("</body></html>");
		//return buf.join(" ");
		liste = buf.join(" ");*/
})();

exports.ret=function() {
	return liste;
};

exports.teams=(function() {
	return teams;
})();

exports.ajout =function (item) {
	//console.log("nouveau: "+item);
	teams.push(item);
	//console.log(teams);
	fs.writeFileSync(file,JSON.stringify(teams,null,4));
	initialize();
};

exports.sup = function(rang) {
	teams.splice(rang,1);
	//console.log(teams);
	fs.writeFileSync(file,JSON.stringify(teams,null,4));
	initialize();
}

exports.getItem = function(rang) {
	return teams[rang];
}

exports.setItem = function(item) {
	team = teams[item.rang];
	team.nom=item.nom;
	team.lieu=item.lieu;
	teams[item.rang]=team;
	fs.writeFileSync(file,JSON.stringify(teams,null,4));
	initialize();
}

function initialize() {
	var buf = [];
	buf.push("<html><head></head><body><h1>liste des teams</h1>");
	buf.push("<table border='1'>");
	buf.push("<tr><th>Action</th><th>Code</th><th>Nom</th><th>Lieu</th></tr>");
		//buf.push(content);
	var t;
	for (i=0; i < teams.length; i++) {
		t = teams[i];
		buf.push("<tr><td><a href='/team/sup/"+i+"'>sup</a> - <a href='/team/mod/"+i+"'>mod</a></td><td>"
			+t.code+"</td><td>"+t.nom+"</td><td>"+t.lieu+"</td></tr>");
			//buf.push(t);
	}
	buf.push("</table>");
	buf.push("<hr><a href='/'>Retour</a>");
	buf.push("<a href='/newteam'>Nouveau Team</a>");
	buf.push("</body></html>");
		//return buf.join(" ");
	liste = buf.join(" ");
};
